hudi-common/src/main/java/org/apache/hudi/common/model/HoodieRecord.java

```diff
@@ -427,6 +428,24 @@ public HoodieRecord rewriteRecordWithNewSchema(Schema recordSchema, Properties p
 
   public abstract Option<Map<String, String>> getMetadata();
 
+  private void initMetadata() {
+    this.metaData = Option.of(new HashMap<>());
```

Comment:
Could we make `this.metaData` final and get rid of this?

Reply:
The constructor in Line 173 does not initialize this metadata, which leaves it as null. Should I fix these constructor instead? Meanwhile, I did meet NPE for metadata previously.